feat : 설정 타입별 값 검증 기능

moonhyeonmin opened the pull request:
## 🛰️ Issue Number

## #18

## **🔗 관련 이슈**

## #9

## **✨ 작업 내용**

- ConfigValidator 클래스에 각 ConfigType별 유효성 검사 메서드를 구현
- PlatformConfigService의 createConfig, updateConfig 메서드에 검증 로직 추가
    - 값이 유효하지 않으면 ValidationException 발생시키고 400 에러
- 정규식 또는 라이브러리를 활용한 엄격한 검증 적용

---

## **✅ 체크리스트**

---

- [x]  코드가 정상적으로 컴파일되나요?
- [x]  테스트 코드를 통과했나요?
- [x]  merge할 브랜치의 위치를 확인했나요?
- [x]  Label을 지정했나요?

## **🎃 새롭게 알게된 사항**

- ConfigValidator 클래스를 구분하여 검증 로직을 분리하는 방법을 실무에서 주로 사용

---

## **📋 참고 사항**

- Task 별로 나눠서 개발했어야 했는데, 해당 사실을 인지하지 못하고 한 번에 커밋해버리는 실수를 저질러버림
    - 일단 API 테스트는 전부 완료해서 PR 올립니다... 만약 다시 해야된다면 댓글로 알려주세요!! 최대한 빠르게 롤백해서 커밋 다시 진행하겠습니다..!

---

- 코드 리뷰 시 논의가 필요한 사항이나 배포 관련 주의 사항을 추가합니다.

moonhyeonmin committed:
feat : 마스킹 관련 로직 삭제

moonhyeonmin commented on src/main/java/com/agenticcp/core/domain/platform/controller/PlatformConfigController.java:
```diff
@@ -20,53 +29,97 @@ public class PlatformConfigController {
-    public ResponseEntity<ApiResponse<PlatformConfig>> getConfigByKey(@PathVariable String configKey) {
+    public ResponseEntity<ApiResponse<PlatformConfigDtos.Response>> getConfigByKey(
+            @PathVariable String configKey) {
         return platformConfigService.getConfigByKey(configKey)
-                .map(config -> ResponseEntity.ok(ApiResponse.success(config)))
+                .map(config -> ResponseEntity.ok(ApiResponse.success(PlatformConfigDtos.Response.of(config))))
```
마스킹 로직은 19번 이슈에서 구현할 것이기 때문에 18번 이슈에는 제외하였습니다~

KwonSunJae commented:
+ 비즈니스 예외 처리 부분은 일단 다 주석 처리해주세요! 
나중에 Issue 열어서 비즈니스 예외 처리하는거 한꺼번에 처리하겠습니다. 
//TODO : BE 
일단 이렇게 표시해놔주세요

moonhyeonmin committed:
TASK: ConfigValidators 유틸리티 클래스 구현
- 키 정책 검증 로직 (길이 3-128, 영문/숫자 시작, 허용문자 제한)
- 타입별 값 검증 로직 (STRING, NUMBER, BOOLEAN, JSON, ENCRYPTED)
- ValidationException을 통한 일관된 오류 처리
- 단위 테스트 포함 (36개 테스트 케이스)

moonhyeonmin committed:
feat : PlatformConfig 엔티티 수정

- isEncrypted, isSystem 필드에 @Builder.Default 적용
- ENCRYPTED 타입일 때 자동으로 isEncrypted=true 설정
- @lombok.EqualsAndHashCode(callSuper = false) 추가

moonhyeonmin committed:
feat : PlatformConfigDtos 구현

- CreateRequest: 생성 시 필수 필드 (key/type/value/isSystem/description)
- UpdateRequest: 수정 시 필드 (type/value/description)
- Response: 조회 응답 (ENCRYPTED 값은 평문으로 노출)
- @Valid 어노테이션을 통한 유효성 검증

moonhyeonmin committed:
feat : PlatformConfigRepository 페이징 메서드 추가

- findAll(Pageable): 전체 설정 페이징 조회
- findByConfigType(ConfigType, Pageable): 타입별 페이징 조회
- findByIsSystem(Boolean, Pageable): 시스템 설정 페이징 조회
- findFilteredConfigs(): 필터링된 설정 페이징 조회
- @Query 어노테이션을 통한 커스텀 쿼리 지원

moonhyeonmin committed:
feat : PlatformConfigService 비즈니스 로직 구현

- 조회: 전체/타입별/시스템 설정 페이징 조회
- 단건: 키로 조회 (Optional 및 예외 처리)
- 생성: 키 정책 및 타입별 값 검증
- 수정: 타입별 값 검증 후 갱신
- 삭제: 시스템 설정 삭제 금지 정책 적용
- @Transactional(readOnly = true) 기본 적용

moonhyeonmin committed:
feat : PlatformConfigController REST API 구현

- GET /api/platform/configs: 전체 설정 페이징 조회 (필터링 지원)
- GET /api/platform/configs/{configKey}: 단건 조회
- GET /api/platform/configs/type/{configType}: 타입별 조회
- GET /api/platform/configs/system: 시스템 설정 조회
- POST /api/platform/configs: 설정 생성 (유효성 검증)
- PUT /api/platform/configs/{configKey}: 설정 수정
- DELETE /api/platform/configs/{configKey}: 설정 삭제
- Swagger UI 문서화 및 @Valid 어노테이션 적용

moonhyeonmin committed:
feat : PlatformConfigService 단위 테스트 구현

- Mockito를 사용한 Repository 계층 Mocking
- 서비스 비즈니스 로직 및 예외 처리 검증
- 16개 테스트 케이스 (조회/생성/수정/삭제/예외처리)
- @ExtendWith(MockitoExtension.class) 적용
- Given-When-Then 패턴 사용

moonhyeonmin committed:
feat : PlatformConfigIntegrationTest 통합 테스트 구현

- @SpringBootTest를 사용한 실제 데이터베이스 연동 테스트
- TestRestTemplate을 통한 HTTP API 호출 테스트
- 전체 플로우 검증 (생성 → 조회 → 수정 → 삭제)
- ENCRYPTED/JSON 타입 설정 테스트
- 시스템 설정 삭제 금지 정책 검증
- 잘못된 타입 값 검증 테스트

moonhyeonmin committed:
feat : PlatformConfigControllerTest 컨트롤러 테스트 구현

- @WebMvcTest를 사용한 컨트롤러 계층 단위 테스트
- MockMvc를 통한 HTTP 요청/응답 테스트
- 모든 REST API 엔드포인트 검증
- 요청/응답 DTO 유효성 검증 테스트
- 예외 처리 및 HTTP 상태 코드 검증
- @MockBean을 통한 서비스 계층 Mocking

moonhyeonmin committed:
feat : 빌드 설정 및 테스트 환경 구성

- pom.xml: Lombok/MapStruct 어노테이션 프로세서 설정
- AgenticCpCoreApplication: @ComponentScan 추가
- application-test.yml: H2 인메모리 DB 테스트 설정

moonhyeonmin commented:
> TASK 기준으로 Commit 다시 정리해주세요! 처음이니까 Rebase 해서 commit 다시 정리하는 법 미리 익혀두면 좋을거 같습니다.
> 
> * Masking 관련 로직은 왜 삭제 했는가 알려주세요

1. TASK 기준으로 Commit 정리 완료했습니다.
2. Masking 관련 로직은 #19 에서 한꺼번에 구현하도록 계획을 짠 줄 알았는데, 두 이슈에서 마스킹을 하는 목표가 다름을 방금 깨달았습니다. 18번 이슈에서 민감값 Masking하는 로직 추가하겠습니다!

moonhyeonmin committed:
feat : 민감 사항 마스킹

moonhyeonmin committed:
feat : 비즈니스 예외 처리 주석

moonhyeonmin commented:
> * 비즈니스 예외 처리 부분은 일단 다 주석 처리해주세요!
>   나중에 Issue 열어서 비즈니스 예외 처리하는거 한꺼번에 처리하겠습니다.
>   //TODO : BE
>   일단 이렇게 표시해놔주세요

- 해당 부분 처리 완료 후 커밋 했습니다!
